### PR TITLE
fix: refactor model definition and model JSON schema

### PR DESCRIPTION
**Pull request by xiaofei-du**
Because

- we want to provide up-to-date specs for all supported model definitions and JSON schema for model-backend related resources. 
- This PR is based on https://github.com/instill-ai/protobufs/pull/67

This commit

- add icons for each model definition
- refactor specs for each model definition
- add JSON schema for model related resources

### CODE REVIEW

File: configs/init/instill/seed/model_definitions.yaml [hunk 1]
```diff
@@ -1,22 +1,131 @@
...
-  model_instance_spec: 
-      repo: 
-      tag: 
\ No newline at end of file
+- id: "github"
+  uid: "909c3278-f7d1-461c-9352-87741bef11d3"
+  title: "GitHub"
+  documentationUrl: "https://docs.instill.tech/integrations/models/github"
+  icon: "github.svg"
+  modelSpecification:
```
Comment (Phelan164) [inline configs/init/instill/seed/model_definitions.yaml:6]: @xiaofei-du should we name same as protobuf for consistency. In protobuf, it named [modelSpec](https://github.com/instill-ai/protobufs/blob/main/instill/model/v1alpha/model_definition.proto#L43) only

File: configs/init/instill/seed/model_definitions.yaml [hunk 2]
```diff
@@ -1,22 +1,131 @@
...
+        description: "The URL of the GitHub repository, e.g. `https://github.com/instill-ai/yolov4`"
+        examples:
+        - "https://github.com/instill-ai/yolov4"
+        - "https://github.com/instill-ai/mobilenetv2"
+        readOnly: true
+        ui_order: 1
+        ui_hidden: true
+        ui_disabled: true
+        ui_component: "text"
+  modelInstanceSpecification:
```
Comment (Phelan164) [inline configs/init/instill/seed/model_definitions.yaml:35]: @xiaofei-du same above

File: configs/init/instill/seed/model_definitions.yaml (hunk 1 above)
Comment (xiaofei-du) [reply configs/init/instill/seed/model_definitions.yaml]: refactored, thanks!

File: configs/init/instill/seed/model_definitions.yaml (hunk 2 above)
Comment (xiaofei-du) [reply configs/init/instill/seed/model_definitions.yaml]: Refactored, thanks!

File: configs/models/model.json [hunk 3]
```diff
@@ -0,0 +1,407 @@
...
+          "ui_order": 6,
+          "ui_hidden": true,
+          "ui_disabled": true
+        },
+        "model_instance_spec": {
+          "type": "object",
+          "title": "Instance specification",
+          "description": "Model definition specification blob. Must be a valid JSON that includes what fields are needed for a model instance",
+          "additionalProperties": false,
+          "ui_order": 6,
```
Comment (Phelan164) [inline configs/models/model.json:103]: @xiaofei-du it should be `"ui_order": 7,` I think

**Comment by Phelan164** [comment]
LGTM

**Review by Phelan164 — APPROVED** [review]

**Pull request merged**